### ISSUE TITLE
"withAuto" components should never render more than once per batch of observed changes

### ISSUE THREAD

**Issue by aleclarson** [alloc/wana#10]
Components wrapped by `withAuto` should never render more than **once** in response to an observable value being changed. 

Unfortunately, when a parent component and its child both observe the same value, the child is rendered up to 2 times (depending on whether or not the child is memoized).

### Solution

1. Make `withAuto` provide a number to descendants that tracks the depth of `withAuto` calls. For example, when a parent and its child are both wrapped by `withAuto`, the child has a depth of 1 and the parent has a depth of 0.

2. When an observed value is changed, add the `withAuto` component to a sorted queue, where components closer to the root come first.

3. Before flushing the sorted queue, keep track of a nonce for each `withAuto` component in the queue. This cached nonce is compared to the current nonce before each `withAuto` component is updated. Only update a component whose current nonce is **equal** to its cached nonce.

In conclusion, `withAuto` will always update parent components first. Child components will avoid re-rendering more than once per batch of observed changes.

**Comment by aleclarson** [comment]
Fixed in `0.7.0`
